### PR TITLE
[poppler][poppler-data] add poppler depends data

### LINKED ISSUE #33756: [poppler] Unable to generate images for Chinese invoices due to not adding a popper data dependency
### Is your feature request related to a problem? Please describe.

Unable to generate images for Chinese invoices due to not adding a popper data dependency
由于没有添加poppler-data依赖导致中文的发票无法生成图片

### Proposed solution

Add this dependency project (I tried to write a port patch, but there were many warnings and I didn't quite understand)
添加一下这个依赖项目（我尝试写了个port补丁，但是好多警告，不太明白）
https://poppler.freedesktop.org/poppler-data-0.4.12.tar.gz

### Describe alternatives you've considered

我尝试过自己写port 应该是可行的，也试过添加一个依赖路径
image
image

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by heweisheng**
come from https://github.com/microsoft/vcpkg/issues/33756
@JonLiu1993 I don't quite understand vcpkg and cmake scripts. Compiling them roughly like this can solve the problem。

Old version cannot display perfectly

the test file
[路径test니그_加水印.pdf](https://github.com/microsoft/vcpkg/files/12605336/test._.pdf)

### CODE REVIEW

**Comment by heweisheng** [comment]
> 请阅读以下贡献者许可协议 （CLA）。如果您同意CLA的观点，请回复以下信息。
> 
> ```
> @microsoft-github-policy-service agree [company="{your company}"]
> ```
> 
> > 选项：
> > 
> > * （默认 - 未指定公司）我对我的提交内容拥有唯一的知识产权，并且我不是在为我的雇主工作的过程中提交内容。
> > 
> > ```
> > @microsoft-github-policy-service agree
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > 
> > * （当公司给出时）我在为我的雇主工作的过程中提交内容（或者我的雇主根据合同或适用法律对我的提交内容拥有知识产权）。我已获得雇主的许可，可以代表我的雇主提交意见并签订本协议。通过在下面签名，定义的术语“您”包括我和我的雇主。
> > 
> > ```
> > @microsoft-github-policy-service agree company="Microsoft"
> > ```
> 
> 贡献者许可协议

@microsoft-github-policy-service agree company="Microsoft"

**Comment by JonLiu1993** [comment]
@heweisheng, Thanks for your PR, this is test error, please take a look:
````
You can move the pkgconfig files with commands similar to:
    file(MAKE_DIRECTORY "${CURRENT_PACKAGES_DIR}/share/pkgconfig")
    file(RENAME "${CURRENT_PACKAGES_DIR}/debug/share/pkgconfig/poppler-data.pc" "${CURRENT_PACKAGES_DIR}/share/pkgconfig/poppler-data.pc")
    vcpkg_fixup_pkgconfig()
    file(REMOVE_RECURSE empty directories left by the above renames)
warning: There should be no absolute paths, such as the following, in an installed package:
    /mnt/vcpkg-ci/packages/poppler-data_x64-linux
    /mnt/vcpkg-ci/installed
    /mnt/vcpkg-ci/buildtrees/poppler-data
    /mnt/vcpkg-ci/downloads
Absolute paths were found in the following files:
    /mnt/vcpkg-ci/packages/poppler-data_x64-linux/debug/share/pkgconfig/poppler-data.pc
    /mnt/vcpkg-ci/packages/poppler-data_x64-linux/share/pkgconfig/poppler-data.pc
error: Found 5 post-build check problem(s). To submit these ports to curated catalogs, please first correct the portfile: /mnt/vss/_work/1/s/ports/poppler-data/portfile.cmake
````

**Comment by JonLiu1993** [comment]
Note: I will be converting your PR to draft status. When you respond, please revert to "ready for review". That way, I can be aware that you've responded since you can't modify the tags.

File: ports/poppler/portfile.cmake [hunk 1]
```diff
@@ -75,6 +74,7 @@ vcpkg_cmake_configure(
         -DCMAKE_REQUIRE_FIND_PACKAGE_Boost=ON
         -DCMAKE_DISABLE_FIND_PACKAGE_GObjectIntrospection=ON
         -DCMAKE_DISABLE_FIND_PACKAGE_GTK=ON
+		"-DPOPPLER_DATADIR=${CURRENT_INSTALLED_DIR}/share/poppler"
```
Comment (dg0yt) [inline ports/poppler/portfile.cmake:77]: I think this injects an absolute path into the binaries. This won't work after deployment.

**Review by dg0yt — COMMENTED** [review]
IMO the data shouldn't become a hard build-time requirement. AFAIU it is needed for "CJK and Cyrillic" support, at run time.

File: ports/poppler/portfile.cmake (hunk 1 above)
Comment (heweisheng) [reply ports/poppler/portfile.cmake:77]: Yes, a more elegant solution is needed

**Comment by heweisheng** [comment]
> IMO 数据不应成为硬性构建时要求。AFAIU 在运行时需要“CJK 和西里尔文”支持。



> @heweisheng, Thanks for your PR, this is test error, please take a look:
> 
> ```
> You can move the pkgconfig files with commands similar to:
>     file(MAKE_DIRECTORY "${CURRENT_PACKAGES_DIR}/share/pkgconfig")
>     file(RENAME "${CURRENT_PACKAGES_DIR}/debug/share/pkgconfig/poppler-data.pc" "${CURRENT_PACKAGES_DIR}/share/pkgconfig/poppler-data.pc")
>     vcpkg_fixup_pkgconfig()
>     file(REMOVE_RECURSE empty directories left by the above renames)
> warning: There should be no absolute paths, such as the following, in an installed package:
>     /mnt/vcpkg-ci/packages/poppler-data_x64-linux
>     /mnt/vcpkg-ci/installed
>     /mnt/vcpkg-ci/buildtrees/poppler-data
>     /mnt/vcpkg-ci/downloads
> Absolute paths were found in the following files:
>     /mnt/vcpkg-ci/packages/poppler-data_x64-linux/debug/share/pkgconfig/poppler-data.pc
>     /mnt/vcpkg-ci/packages/poppler-data_x64-linux/share/pkgconfig/poppler-data.pc
> error: Found 5 post-build check problem(s). To submit these ports to curated catalogs, please first correct the portfile: /mnt/vss/_work/1/s/ports/poppler-data/portfile.cmake
> ```

let me try

**Comment by JonLiu1993** [comment]
Ping @heweisheng for response.

**Comment by heweisheng** [comment]
> Ping @heweisheng for response.

I don't know how to change it will be better,such as @dg0yt say

**Pull request closed without merging**